### PR TITLE
Add applications sync to application set sync policies

### PR DESCRIPTION
**Pull request by ahublersos**
Fixes https://github.com/oboukili/terraform-provider-argocd/issues/333.
Adds support for applications_sync to the application set resource.

I also nearly installed the test manifests into a live cluster by running `testacc_prepare_env` so I added a few safety safety nets to prevent that from happening in the future. Happy to split this out into a new PR if that's preferred.

### CODE REVIEW

**Comment by ahublersos** [comment]
I don't know why the `TestAccArgoCDCluster_invalidSameServer` test is failing. That passed in local testing and doesn't seem related to changes in the application_set resource.

**Comment by onematchfox** [comment]
> I don't know why the `TestAccArgoCDCluster_invalidSameServer` test is failing. That passed in local testing and doesn't seem related to changes in the application_set resource.

Yeah, that's just a flaky test. Everything passing now. Will try take a look at the changes a some point today/tomorrow morning.

File: argocd/structure_application_set.go [hunk 1]
```diff
@@ -12,14 +12,14 @@ import (
 	meta "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
-func expandApplicationSet(d *schema.ResourceData, featureMultipleApplicationSourcesSupported bool) (metadata meta.ObjectMeta, spec application.ApplicationSetSpec, err error) {
+func expandApplicationSet(d *schema.ResourceData, featureMultipleApplicationSourcesSupported bool, featureApplicationSetApplicationsSyncPolicy bool) (metadata meta.ObjectMeta, spec application.ApplicationSetSpec, err error) {
```
Comment (onematchfox) [inline argocd/structure_application_set.go:15]: I realise this is a bit confusing, but can you please drop this parameter (and it's usage in this file) and rather use the feature flag to implement a check on the user's provided configuration as per https://github.com/oboukili/terraform-provider-argocd/blob/d84d09101f8c1954cf613df69890a846540692b7/argocd/resource_argocd_application_set.go#L49-L51.

In essence, this is the pattern used by all other feature flags - we check the user's configuration of the Terraform resource and return an error if they try to configure something via Terraform that isn't supported by their version of ArgoCD (thereby preventing the configuration from being applied to the resulting CRUD call the provider makes). Multiple applications sources is the exception here in that we support versions that both have and do not have the feature and thus, need [branching logic](https://github.com/oboukili/terraform-provider-argocd/blob/d84d09101f8c1954cf613df69890a846540692b7/argocd/structure_application_set.go#L800-L807) in what we set on the CRUD call.

**Review by onematchfox — CHANGES_REQUESTED** [review]
Sorry for the slow feedback. Hopefully just one small change - otherwise LGTM.

File: argocd/structure_application_set.go (hunk 1 above)
Comment (ahublersos) [reply argocd/structure_application_set.go]: Got it, thanks for the explanation! I've removed this parameter and added a feature flag check in the resource.

**Comment by onematchfox** [comment]
This isn't stale. I just need to get around to looking at the failing tests.

**Review by onematchfox — APPROVED** [review]

**Comment by onematchfox** [comment]
Sorry for the delay here @ahublersos and thanks for your patience!

**Comment by onematchfox** [comment]
@ahublersos can/should I go ahead and cut a new release now?

**Comment by ahublersos** [comment]
@onematchfox yes, please do!

**Comment by onematchfox** [comment]
> @onematchfox yes, please do!

Sorry... more delays. Our release pipeline is broken. Will try upgrade/fix this weekend.

**Pull request merged**